portal-impl/src/com/liferay/portlet/announcements/service/impl/AnnouncementsEntryServiceImpl.java

```
@@ -56,23 +56,6 @@ public AnnouncementsEntry addEntry(
 
 		PermissionChecker permissionChecker = getPermissionChecker();
 
-		if (alert) {
-			String portletId = PortletProviderUtil.getPortletId(
-				AnnouncementsConstants.CLASS_NAME_ALERTS_ENTRY,
-				PortletProvider.Action.MANAGE);
-
-			AnnouncementsEntryPermission.check(
-				permissionChecker, plid, portletId, ActionKeys.ADD_ENTRY);
-		}
-		else {
-			String portletId = PortletProviderUtil.getPortletId(
-				AnnouncementsEntry.class.getName(),
-				PortletProvider.Action.MANAGE);
-
-			AnnouncementsEntryPermission.check(
-				permissionChecker, plid, portletId, ActionKeys.ADD_ENTRY);
-		}
-
```

Hey @topolik yeah, I think that you can throw an exception if the classname is not any of the supported. Note that we also support classnameId 0 for general announcements as in https://github.com/stian-sigvartsen/liferay-portal/blob/9d9d8a288fe1313647031a5f992daa125da11718/portal-impl/src/com/liferay/portlet/announcements/service/impl/AnnouncementsEntryServiceImpl.java#L59